src/ol/format/kml.js

```diff
@@ -1078,6 +1108,9 @@ ol.format.KML.readStyle_ = function(node, objectStack) {
   var imageStyle = /** @type {ol.style.Image} */
       ('imageStyle' in styleObject ?
           styleObject['imageStyle'] : ol.format.KML.DEFAULT_IMAGE_STYLE_);
+  if (imageStyle == ol.format.KML.DEFAULT_NO_IMAGE_STYLE_) {
```

Comment:
There are acutally 3 states: One is undefined, one is explicit where the
Icon block is included but no icon is defined because the user wants no
icon to be drawn; and the third is defined.  So this DEFAULT_NO_IMAGE_STYLE
flag is to indicate explicitly no image.  We *could* have null mean one
thing and undefined mean another, but I thought that calling out NO_IMAGE
would be more clear, and it's only when the KML is being loaded that this
flag is used.

On Sun, Aug 21, 2016 at 11:29 AM, Andreas Hocevar <notifications@github.com>
wrote:

> In src/ol/format/kml.js
> <https://github.com/openlayers/ol3/pull/5745#discussion_r75600789>:
>
> > @@ -1078,6 +1108,9 @@ ol.format.KML.readStyle_ = function(node, objectStack) {
> >    var imageStyle = /** @type {ol.style.Image} */
> >        ('imageStyle' in styleObject ?
> >            styleObject['imageStyle'] : ol.format.KML.DEFAULT_IMAGE_STYLE_);
> > +  if (imageStyle == ol.format.KML.DEFAULT_NO_IMAGE_STYLE_) {
>
> Couldn't you initialize the ol.format.ol.KML.DEFAULT_NO_IMAGE_STYLE_ with
> undefined instead? Then you wouldn't need this if-block.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openlayers/ol3/pull/5745/files/e0d75555c5a110b24fe63e618966ac73993c198c#r75600789>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFgK8OymgASl7dm03uMQskWfNikfryOeks5qiJkSgaJpZM4Jm_9w>
> .
>